src/utils/utils.cpp

```diff
@@ -584,12 +584,19 @@ PetscErrorCode ExaGOInitialize(MPI_Comm comm, int *argc, char ***argv,
 PetscErrorCode ExaGOFinalize() {
   PetscFunctionBegin;
   ExaGOLog(EXAGO_LOG_INFO, "Finalizing {} application.", ExaGOCurrentAppName);
+  MPI_Comm comm = MPI_COMM_WORLD;
+  int my_rank;
+  int logging_rank = 0;
+  auto err = MPI_Comm_rank(comm, &my_rank);
```

Comment:
Still remnant rank stuff to refactor here.

Reply:
Again, this falls outside of the FunctionalityTestContext. The better way to handle this would have been to create an  environment object, initialize rank and nprocs inside that and then just access the values whenever you need them. Since these are just function calls, that doesn't work. I'd just leave it. If you want, you could get rid of the actual check on the error since if the code made it this far it is extremely unlikely that MPI_Comm_rank will fail.